aptos-move/block-executor/src/executor.rs

```diff
@@ -1271,81 +1278,87 @@ where
+            base_view,
+            false,
+        );
+
+        // If sequential gave us result, return it
+        let sequential_error = match sequential_result {
+            Ok(output) => {
+                return Ok(output);
+            },
+            Err(SequentialBlockExecutionError::ResourceGroupSerializationError) => {
+                if !self.config.local.allow_fallback {
+                    panic!("Parallel execution failed and fallback is not allowed");
                 }
-            }
 
-            if resource_group_bcs_fallback {
-                alert!("Resource group serialization fallback");
+                // All logs from the first pass of sequential execution should be cleared and not reported.
+                // Clear by re-initializing the speculative logs.
                 init_speculative_logs(signature_verified_block.len());
-                ret = self.execute_transactions_sequential(
+
+                let sequential_result = self.execute_transactions_sequential(
                     executor_arguments,
                     signature_verified_block,
                     base_view,
                     true,
                 );
-            }
 
-            // If after trying available fallbacks, we still are askign to do a fallback,
-            // something unrecoverable went wrong.
-            if let Err(BlockExecutionError::FallbackToSequential(e)) = &ret {
-                // TODO[agg_v2][fix] make sure this can never happen - we have sequential raising
-                // this error often when something that should never happen goes wrong
-                panic!("Sequential execution failed with {:?}", e);
-            }
-        } else if let Err(BlockExecutionError::FallbackToSequential(e)) = &ret {
-            // TODO[agg_v2][fix] make sure this can never happen - we have sequential raising
-            // this error often when something that should never happen goes wrong
-            panic!("Fallback disabled, execution failed due to {:?}", e);
+                // If sequential gave us result, return it
+                match sequential_result {
+                    Ok(output) => {
+                        return Ok(output);
+                    },
+                    Err(SequentialBlockExecutionError::ResourceGroupSerializationError) => {
+                        BlockExecutionError::FatalBlockExecutorError(code_invariant_error(
+                            "resource group serialization during bcs fallback should not happen",
+                        ))
+                    },
+                    Err(SequentialBlockExecutionError::ErrorToReturn(err)) => err,
+                }
+            },
+            Err(SequentialBlockExecutionError::ErrorToReturn(err)) => err,
+        };
+
+        // if not skip failed blocks enabled
+        if self.config.local.discard_failed_blocks {
+            // We cannot execute block, disard everything (including block metadata and validator transactions)
```

Comment:
typo: discard.

Is it safe to discard blockmetadata or some other transactions? (can they have some specialized logic that wouldn't be in the next block?)